src/Microsoft.HttpRepl/Commands/ChangeDirectoryCommand.cs

```diff
@@ -57,17 +57,26 @@ protected override Task ExecuteAsync(IShellState shellState, HttpState programSt
                 {
                     IDirectoryStructure s = programState.Structure.TraverseTo(programState.PathSections.Reverse());
 
-                    if (s.RequestInfo == null || s.RequestInfo.Methods.Count == 0)
+                    string thisDirMethod;
+
+                    // If there's no RequestInfo/Methods AND there's no (sub)DirectoryNames, it must be an
+                    // autogenerated directory 
+                    if ((s.RequestInfo?.Methods is null || s.RequestInfo.Methods.Count == 0) && (s.DirectoryNames is null || !s.DirectoryNames.Any()))
```

Comment:
instead of a comment explaining the long predicate, how about:

```
bool hasNoRequestInfo = s.RequestInfo?.Methods is null || s.RequestInfo.Methods.Count == 0;
bool hasNoDirectoryNames = s.DirectoryNames is null || !s.DirectoryNames.Any();
if(hasNoRequestInfo && hasNoDirectoryNames)
{ /* ... */ }
```

Reply:
So the comment wasn't as much about explaining the predicate as making our assumption (that the lack of those two things meant it was auto-generated) explicit rather than implicit by the code. I like the way you listed it better anyway, though, so I changed it and modified the comment a little bit in https://github.com/aspnet/HttpRepl/commit/afc0086f7d73c95c514a5fc9defe7f89d80f4a76.